pkg/commands/git.go

```diff
@@ -184,8 +186,9 @@ func (c *GitCommand) GetStatusFiles() []*File {
 			HasUnstagedChanges:      unstagedChange != " ",
 			Tracked:                 !untracked,
 			Deleted:                 unstagedChange == "D" || stagedChange == "D",
-			HasMergeConflicts:       change == "UU" || change == "AA" || change == "DU",
-			HasInlineMergeConflicts: change == "UU" || change == "AA",
+			HasMergeConflicts:       hasMergeConflicts,
+			HasInlineMergeConflicts: hasInlineMergeConflicts,
+			NeedReset:               !hasNoStagedChanges || hasMergeConflicts || hasInlineMergeConflicts,
```

Comment:
given that this can be derived from those three variables, which are accessible on the File, I would not store it here, and instead just check those three fields when we need to